src/sqlfluff/rules/L009.py

```diff
@@ -8,7 +8,27 @@
 
 @document_fix_compatible
 class Rule_L009(BaseRule):
-    """Files must end with a trailing newline."""
+    r"""Files must end with a trailing newline.
+
+    | **Anti-pattern**
+    | The content in file without end with a trailing newline.
+
+    .. code-block:: sql
+
+        SELECT
+            a
+        FROM foo
+
+    | **Best practice**
+    | Add trailing newline to the end, the \n character represents a newline symbol.
+
+    .. code-block:: sql
+
+        SELECT
+            a
+        FROM foo\n
```

Comment:
Perhaps include some of the more common mistakes I often see?

I also suggest changing from using \n (to represent newline) to $ (to represent end of file) as \n could be confusing as to which line it appears on.

```suggestion
    | **Anti-pattern**
    | The content in file without ends without a trailing newline, the $ represents end of file.

    .. code-block:: sql

        SELECT
            a
        FROM foo$
        
        -- Ending on an indented line means there is no newline at the end of the file, the • represents space.

        SELECT
        ••••a
        FROM
        ••••foo
        ••••$

        -- Ending on a semi-colon means the last line is not a newline.

        SELECT
            a
        FROM foo
        ;$

    | **Best practice**
    | Add trailing newline to the end, the $ character represents end of file.

    .. code-block:: sql

        SELECT
            a
        FROM foo
        $
        
        -- Ensuring the last line is not indented so is just a newline.

       SELECT
        ••••a
        FROM
        ••••foo
        $

        -- Even when ending on a semi-colon, ensure there is a newline after

        SELECT
            a
        FROM foo
        ;
        $
```

Reply:
What a good example for common mistake! 👍